spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/core/PubSubConfigurationTests.java

```diff
@@ -112,13 +112,42 @@ public void testSubscriberRetrySettings() {
 	}
 
 	@Test
-	public void testSubscriberMapProperties_default() {
+	public void testSubscriberMapProperties_defaultOrGlobal_addToMap() {
 		PubSubConfiguration pubSubConfiguration = new PubSubConfiguration();
 
-		assertThat(pubSubConfiguration.getSubscriber("subscription-name").getExecutorThreads())
-				.isEqualTo(4);
-		assertThat(pubSubConfiguration.getSubscription().get("subscription-name").getExecutorThreads())
-				.isEqualTo(4);
+		assertThat(pubSubConfiguration.getSubscription()).isEmpty();
+		assertThat(pubSubConfiguration.getSubscriber("subscription-name").getExecutorThreads()).isEqualTo(4);
+		assertThat(pubSubConfiguration.getSubscription()).hasSize(1);
+		assertThat(pubSubConfiguration.getSubscription().get("subscription-name").getExecutorThreads()).isEqualTo(4);
+	}
+
+	@Test
+	public void testSubscriberMapProperties_returnCustom() {
+		PubSubConfiguration pubSubConfiguration = new PubSubConfiguration();
+		PubSubConfiguration.Subscriber subscriber = new PubSubConfiguration.Subscriber();
+		subscriber.setExecutorThreads(8);
+
+		pubSubConfiguration.getSubscription().put("subscription-name", subscriber);
```

Comment:
(non-actionable) So, this was handy for testing, but returning mutable collections from code is generally bad for this exact reason. Ideally, all maps/lists/etc that come from getters should be turned immutable with `ImmutableMap.of()`, `ImmutableList.of()` etc.

However, Spring Boot itself is full of this pattern ([example](https://github.com/spring-projects/spring-boot/blob/c406dda18160a29649eef6bbb73f3c93674f4028/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceProperties.java#L558)) -- properties are numerous, and only loaded at context startup, so I guess it's a pretty safe use in real life.